### PR TITLE
add serialization test to losses

### PR DESCRIPTION
**Pull request by LukeWood**
fixes https://github.com/keras-team/keras-cv/issues/1347

### CODE REVIEW

File: keras_cv/losses/serialization_test.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import inspect
+
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_cv import core
+from keras_cv import losses as cv_losses
+
+
+def exhaustive_compare(obj1, obj2):
```
Comment (ianstenbit) [inline keras_cv/losses/serialization_test.py:23]: Instead of copy-pastaing this, let's make it a shared test util.

File: keras_cv/losses/serialization_test.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        ("GIoULoss", cv_losses.GIoULoss, {"bounding_box_format": "xywh"}),
+        (
+            "BinaryPenaltyReducedFocalCrossEntropy",
+            cv_losses.BinaryPenaltyReducedFocalCrossEntropy,
+            {},
+        ),
+        ("SimCLRLoss", cv_losses.SimCLRLoss, {"temperature": 0.5}),
+        ("SmoothL1Loss", cv_losses.SmoothL1Loss, {}),
+    )
+    def test_layer_serialization(self, loss_cls, init_args):
```
Comment (ianstenbit) [inline keras_cv/losses/serialization_test.py:102]: test_loss_serialization

File: keras_cv/losses/serialization_test.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        config = loss.get_config()
+        self.assertAllInitParametersAreInConfig(loss_cls, config)
+
+        reconstructed_loss = loss_cls.from_config(config)
+
+        self.assertTrue(
+            config_equals(loss.get_config(), reconstructed_loss.get_config())
+        )
+
+    def assertAllInitParametersAreInConfig(self, layer_cls, config):
```
Comment (ianstenbit) [inline keras_cv/losses/serialization_test.py:113]: loss_cls

**Review by ianstenbit — CHANGES_REQUESTED** [review]

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Maybe my comments got buried but it looks like they're still unresolved. Maybe I'm missing something?

**Comment by LukeWood** [comment]
> Maybe my comments got buried but it looks like they're still unresolved. Maybe I'm missing something?

Woah weird - I only saw the loss_cls one.  Will fix the others.

File: keras_cv/losses/serialization_test.py (hunk 1 above)
Comment (LukeWood) [reply keras_cv/losses/serialization_test.py]: Done

File: keras_cv/utils/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+import inspect
+
+import tensorflow as tf
+
+from keras_cv import core
+from keras_cv import layers as cv_layers
+
+
+def exhaustive_compare(obj1, obj2):
```
Comment (ianstenbit) [inline keras_cv/utils/test_utils.py:23]: A docstring would be useful here

File: keras_cv/utils/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    classes_supporting_get_config = (
+        core.FactorSampler,
+        tf.keras.layers.Layer,
+        cv_layers.BaseImageAugmentationLayer,
```
Comment (ianstenbit) [inline keras_cv/utils/test_utils.py:27]: Is this redundant, since it subclasses Layer?

Id it's not redundant, and we need it, do we also need to list `Loss` here?

**Review by ianstenbit — APPROVED** [review]
LG mod comments
